### PR TITLE
Schedules are messing with each other due to bad management of groups

### PR DESCRIPTION
**Pull request by estellecomment**
# Description

Silence schedules one by one rather than all together, to fix a bug where schedules' clearing was influencing other schedules' clearing. 
See test that fails because of bug : https://github.com/medic/medic-sentinel/pull/163/commits/bca9411c64e3c87a3284b5f314079a11b8c3dae6

medic/medic-webapp#3824

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.
- [ ] Webapp CI runs clean against this branch

### CODE REVIEW

**Comment by estellecomment** [comment]
There's a lot of refactoring, so reviewing will be tricky. 
 - rename vars and funcs
 - remove the `options` argument to functions, because it's hard to tell what's in it, so the code is hard to read and reuse, and some things are passed which actually aren't needed.
 - de-expose functions : only expose functions in `modules.exports` if they're used outside of the module. This caused changes to the tests, because some exposed functions were exposed to be mocked for testing, in order to test sub-parts of the module. I changed most of these tests to test the whole flow instead. (`_silenceReminders` is left exposed, too much changes needed)
 - split the code into functions that are sub-routines, instead of functions that are chopping a large function into first-part-second-part-third-part.
 - ES6 : remove some use of underscore, and add `const` and `=>`
 - create vars where needed, rather than at top of function.

I suggest reviewing in 3 steps : 
 - refactoring commits : https://github.com/medic/medic-sentinel/pull/163/files/57d124610e7f0b91ed4cd56577a26c3cd2fb9e7d
 - adding of the failing test (second-to-last commit): https://github.com/medic/medic-sentinel/pull/163/commits/bca9411c64e3c87a3284b5f314079a11b8c3dae6
 - fixing of the code, making the test pass (last commit) : https://github.com/medic/medic-sentinel/pull/163/commits/34f1467816fc3943fa1e09e9b7b35011d18672d2

File: lib/utils.js [hunk 1]
```diff
@@ -304,6 +304,56 @@ const getPatient = (db, patientShortcodeId, includeDocs, callback) => {
   });
 };
 
+/*
+ * type is a string
+ */
+const getScheduledTasksByType = (doc, type) => {
+  const scheduled_tasks = doc && doc.scheduled_tasks;
+  return scheduled_tasks.filter(task => task.type === type );
```
Comment (mandric) [inline lib/utils.js]: This will throw if scheduled_tasks is undefined or null.

**Comment by mandric** [comment]
Might be good to do some acceptance testing on this one since it's a decent sized refactor and kind of a delicate part of the code.

**Comment by estellecomment** [comment]
Force-pushed with squashed commits for clean merge. (so links above are broken)

**Pull request merged**